src/nomad_measurements/utils.py

```diff
@@ -106,3 +109,21 @@ def merge_sections(
                 logger.warning(warning)
             else:
                 print(warning)
+
+
+def to_pint_quantity(value: Any=None, unit: str=None) -> Any:
+    '''
+    Attempts to generate a pint quantity based on whether value or/and unit are available.
+
+    Args:
+        value (Any): Value of the quantity.
+        unit (str): Unit of the quantity.
+
+    Returns:
+        Any: Processed quantity with datatype depending on the value.
+    '''
+    if not unit:
+        if isinstance(value, str):
+            return value
+        return value * ureg('dimensionless')
```

Comment:
This will currently not work for metainfo quantities without unit. I would skip this for now and just merge it.

Reply:
Alright, we can come back to this in a later PR.